src/controller/AbstractDnssdDiscoveryController.cpp

```diff
@@ -35,7 +35,8 @@ void AbstractDnssdDiscoveryController::OnNodeDiscovered(const chip::Dnssd::Disco
             continue;
         }
         if (strcmp(discoveredNode.resolutionData.hostName, nodeData.resolutionData.hostName) == 0 &&
-            discoveredNode.resolutionData.port == nodeData.resolutionData.port)
+            discoveredNode.resolutionData.port == nodeData.resolutionData.port &&
+            discoveredNode.resolutionData.ipAddress == nodeData.resolutionData.ipAddress)
```

Comment:
This doesn't look right to me. `ipAddress` is an array

Reply:
@joonhaengHeo ^

Reply:
Created https://github.com/project-chip/connectedhomeip/issues/32576 and assigned it to you @joonhaengHeo